ui/src/gist.rs

```diff
@@ -0,0 +1,93 @@
+use hubcaps::gists::{self, Content, GistOptions};
+use hubcaps::{Credentials, Github};
+use std::collections::HashMap;
+use tokio_core::reactor::{Core, Handle};
+use hyper;
+use hyper_tls;
+
+const FILENAME: &str = "playground.rs";
+const DESCRIPTION: &str = "Code shared from the Rust Playground";
+
+pub struct Gist {
+    pub id: String,
+    pub url: String,
+    pub code: String,
+}
+
+impl From<gists::Gist> for Gist {
+    fn from(other: gists::Gist) -> Self {
+        let mut files: Vec<_> = other.files
+            .into_iter()
+            .map(|(name, file)| (name, file.content.unwrap_or_default()))
+            .collect();
+
+        files.sort_by(|&(ref name1, _), &(ref name2, _)| name1.cmp(name2));
+
+        let code = match files.len() {
+            0 | 1 => files.into_iter().map(|(_, content)| content).collect(),
+            _ => {
+                files
+                    .into_iter()
+                    .map(|(name, content)| format!("// ${}\n\n${}\n\n", name, content))
+                    .collect()
+            }
+        };
+
+        Gist {
+            id: other.id,
+            url: other.html_url,
+            code: code,
+        }
+    }
+}
+
+pub fn create(token: String, code: String) -> Gist {
+    let mut core = core();
+    let github = github(token, &core.handle());
+
+    let file = Content {
+        filename: None,
+        content: code,
+    };
+
+    let mut files = HashMap::new();
+    files.insert(FILENAME.into(), file);
```

Comment:
Hm, I'm probably missing some context around how gists are represented here, in hubcaps, and on github, but it seems a little strange that you have a filename, and are using it here on line 54, but on line 49 the `hubcaps::gists::Content` gets `filename: None`?